fixture-monkey-api/src/main/java/com/navercorp/fixturemonkey/api/random/RegexGenerator.java

```diff
@@ -18,80 +18,97 @@
 
 package com.navercorp.fixturemonkey.api.random;
 
-import static java.util.stream.Collectors.toList;
-
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import javax.annotation.Nullable;
 
 import org.apiguardian.api.API;
 import org.apiguardian.api.API.Status;
 
-import com.mifmif.common.regex.Generex;
-
-import dk.brics.automaton.RegExp;
+import com.github.curiousoddman.rgxgen.RgxGen;
+import com.github.curiousoddman.rgxgen.config.RgxGenOption;
+import com.github.curiousoddman.rgxgen.config.RgxGenProperties;
 
 @API(since = "0.6.9", status = Status.MAINTAINED)
 public final class RegexGenerator {
-	private static final Map<String, String> PREDEFINED_CHARACTER_CLASSES;
-
-	static {
-		Map<String, String> characterClasses = new HashMap<>();
-		characterClasses.put("\\\\d", "[0-9]");
-		characterClasses.put("\\\\D", "[^0-9]");
-		characterClasses.put("\\\\s", "[ \t\n\f\r]");
-		characterClasses.put("\\\\S", "[^ \t\n\f\r]");
-		characterClasses.put("\\\\w", "[a-zA-Z_0-9]");
-		characterClasses.put("\\\\W", "[^a-zA-Z_0-9]");
-		PREDEFINED_CHARACTER_CLASSES = Collections.unmodifiableMap(characterClasses);
-	}
+	private static final int FLAG_CASE_INSENSITIVE = 2;
+	public static final int DEFAULT_REGEXP_GENERATION_TIMEOUT_SEC = 10;
+	public static final int DEFAULT_REGEXP_GENERATION_MAX_SIZE = 100;
 
 	public List<String> generateAll(String regex, int[] flags, @Nullable Integer min, @Nullable Integer max) {
```

Comment:
I think it is better to deprecate `generateAll`.
It leads to performance degradation.
New API required for the following situation.

https://github.com/naver/fixture-monkey/blob/22fcdadc701224c88281fb7e2858793c049bb992/fixture-monkey-api/src/main/java/com/navercorp/fixturemonkey/api/jqwik/JqwikJavaArbitraryResolver.java#L84-L84

Reply:
I agree that performance degradation might be an issue here.

Do you have a specific direction in mind? I'm curious whether you are considering a direction that completely avoids supporting pattern validation

Reply:
Deprecated `generateAll` and made a method `generate` that only generates one string. It is called lazily with `Arbitraries.ofSuppliers()`.